### ISSUE TITLE
How to build the docker image from the source code

### ISSUE THREAD

**Issue by PritomAhmed** [uber/jaeger#237]
Hi,

I want to build the docker image to test some modifications I make in the code bas of jaeger. I followed the steps mention in  the [guidelines for contributing](https://github.com/uber/jaeger/blob/master/CONTRIBUTING.md).

Now I want to build the docker so that I can deploy it. How can I do that ? I already have tried the following:
1) Running the script(s) in /travis folder
2) try "docker-compose up" in the cross dock folder
3) Try the "docker build" command using the Dockerfile in the cmd/standalone 

But none of them seems to work. I think I am missing something trivial.

**Comment by black-adder** [comment]
```
docker build -f cmd/standalone/Dockerfile -t <INSERTNAME> .
docker run -d -p 16686:16686 <INSERTNAME>
```
This will build the docker image and run it locally.

**Comment by PritomAhmed** [comment]
Thanks for the quick reply.

But running the first command, I have the following error

_Step 8/10 : COPY ./jaeger-uai-build /go/src/jaeger-ui-build
lstat jaeger-ui-build: no such file or directory_

Now it seems that "**jaeger-ui-build**" is not built properly or at all. Should I run some other command before running docker build ?

 So far I ran the following commands before running _docker build_

**git submodule update --init --recursive
glide install
make test**

**Comment by yurishkuro** [comment]
The docker images are built by `./travis/build-docker-images.sh`

But you may want to first comment out the last line in that script that calls `./travis/upload-to-docker.sh`, since that step will fail

**Comment by PritomAhmed** [comment]
By running `./travis/build-docker-images.sh,` I get 

> skip docker upload for PR
which means that I am not running from the master branch. But I am.
Anything other after line 7 in sh file is not executed.

**Comment by yurishkuro** [comment]
ok, well I think this is the meat of that script:

```
source ~/.nvm/nvm.sh
nvm use 6
DOCKER_NAMESPACE=jaegertracing DOCKER_TAG=${COMMIT} make docker
```

**Comment by PritomAhmed** [comment]
I commented out the checking for BRANCH in the script (build-docker-images.sh) and the script worked for some time but at the end it gives the following error:

> invalid argument "jaegertracing/jaeger-cassandra-schema:" for t: Error parsing reference: "jaegertracing/jaeger-cassandra-schema:" is not a valid repository/tag: invalid reference format
> See 'docker build --help'.
> invalid argument "jaegertracing/jaeger-agent:" for t: Error parsing reference: "jaegertracing/jaeger-agent:" is not a valid repository/tag: invalid reference format
> See 'docker build --help'.
> invalid argument "jaegertracing/jaeger-collector:" for t: Error parsing reference: "jaegertracing/jaeger-collector:" is not a valid repository/tag: invalid reference format
> See 'docker build --help'.
> invalid argument "jaegertracing/jaeger-query:" for t: Error parsing reference: "jaegertracing/jaeger-query:" is not a valid repository/tag: invalid reference format
> See 'docker build --help'.
> make: *** [docker] Error 125


Seems to me that it can not find

> jaegertracing/jaeger-cassandra-schema
> jaegertracing/jaeger-agent
> jaegertracing/jaeger-collector
> jaegertracing/jaeger-query
>

**Comment by yurishkuro** [comment]
it's probably from `DOCKER_TAG=${COMMIT}`, you may want to assign some fixed value to the DOCKER_TAG

**Comment by PritomAhmed** [comment]
Thanks. It worked. How can I find the docker image and run it locally ?

**Comment by jpkrohling** [comment]
The appropriate way to build all images is to run `make docker`. The images are built locally and available at the local Docker repository. The target `make docker-push` will push to Docker Hub.

In both cases, you can override `DOCKER_NAMESPACE` to specify which namespace it should use. The default value is `$USER`, which is suitable for local builds. On Travis, this value is set to jaegertracing.

**Comment by jpkrohling** [comment]
@yurishkuro , please assign this issue to me. I'll improve the appropriated docs (`CONTRIBUTING.md` and possibly others) with the instructions.

**Comment by PritomAhmed** [comment]
It worked. Here is what I did.
1) I changed the ${COMMIT} parts with a fixed value in _build-all-in-one-image.sh_ file
```
./travis/build-all-in-one-image.sh
```
OR
run the following command
```
docker build -f cmd/standalone/Dockerfile -t <repo_name> .
```
Either has the same effect.

2) Check if the image is build using `docker image`. And then run the following command,
```
docker run -d -p5775:5775/udp -p6831:6831/udp -p6832:6832/udp   -p5778:5778 -p16686:16686 -p14268:14268 <repo_name>:latest
```

If we want to make the hotrod example working locally, we have to rebuild it and run using the same commands that are shown in the [tutorial](https://medium.com/opentracing/take-opentracing-for-a-hotrod-ride-f6e3141f7941) and the trace system will work perfectly from a local build.

Thanks @yurishkuro  and @black-adder for your help. I am closing this issue.
